compiler/src/Language/Bond/Codegen/TypeMapping.hs

```diff
@@ -329,6 +334,17 @@ aliasTypeName a args = do
     fragment (Fragment s) = pureText s
     fragment (Placeholder i) = typeName $ args !! i
 
+-- | Builder for the type alias element name in context of 'TypeNameBuilder' monad.
+aliasElementTypeName :: Declaration -> [Type] -> TypeNameBuilder
+aliasElementTypeName a args = do
```

Comment:
The only difference I see between `aliasElementTypeName` and `aliasTypeName` is that the former used `elementTypeName` and the latter uses `typeName`. Consider abstracting this out, particularly given how long this function is.